autoconfig.json

```diff
@@ -1,34 +1,119 @@
+        ]
+      },
+      "DelegatedConfig": {
+        "Read": [
+          "/routing/v1/providers",
+          "/routing/v1/peers",
+          "/routing/v1/ipns"
+        ],
+        "Write": []
+      }
+    },
+    "IPNI": {
+      "URL": "https://cid.contact",
+      "Description": "Network Indexer - content routing database for large storage providers",
+      "DelegatedConfig": {
+        "Read": [
+          "/routing/v1/providers"
+        ],
+        "Write": []
+      }
+    },
+    "IPNS": {
+      "URL": "https://specs.ipfs.tech/ipns/ipns-record/",
+      "Description": "InterPlanetary Name System - signed mutable records that point to CIDs with optional metadata",
+      "DelegatedConfig": {
+        "Read": [
+          "/routing/v1/ipns"
+        ],
+        "Write": [
+          "/routing/v1/ipns"
+        ]
+      }
+    },
+    "DNS": {
+      "URL": "https://datatracker.ietf.org/doc/html/rfc8484",
+      "Description": "DNS over HTTPS (DoH) - protocol for performing DNS resolution via the HTTPS protocol",
+      "DelegatedConfig": {
+        "Read": [
+          "/dns-query"
+        ],
+        "Write": []
+      }
+    },
+    "Example": {
+      "URL": "https://example.com",
+      "Description": "Example routing system demonstrating extensibility. Clients should ignore unknown APIs they don't recognize",
+      "DelegatedConfig": {
+        "Read": [
+          "/example/v0/read"
+        ],
+        "Write": [
+          "/example/v0/write"
+        ]
+      }
+    }
+  },
   "DNSResolvers": {
     "eth.": [
       "https://dns.eth.limo/dns-query",
       "https://dns.eth.link/dns-query"
     ]
   },
-  "DelegatedRouters": {
-    "mainnet-for-nodes-with-dht": [
-      "https://cid.contact/routing/v1/providers"
-    ],
-    "mainnet-for-nodes-without-dht": [
-      "https://delegated-ipfs.dev/routing/v1/providers",
-      "https://delegated-ipfs.dev/routing/v1/peers",
-      "https://delegated-ipfs.dev/routing/v1/ipns"
-    ]
+  "DelegatedEndpoints": {
+    "https://cid.contact": {
+      "Systems": ["IPNI"],
+      "Read": [
+        "/routing/v1/providers"
+      ],
+      "Write": []
+    },
+    "https://delegated-ipfs.dev": {
+      "Systems": ["AminoDHT", "IPNI", "IPNS", "DNS"],
+      "Read": [
+        "/routing/v1/providers",
+        "/routing/v1/peers",
+        "/routing/v1/ipns",
+        "/dns-query"
+      ],
+      "Write": [
+        "/routing/v1/ipns"
+      ]
+    },
+    "https://example.com": {
+      "Systems": ["Example"],
+      "Read": [
+        "/example/v0/read"
+      ],
+      "Write": [
+        "/example/v0/write"
+      ]
+    }
   },
-  "DelegatedPublishers": {
-    "mainnet-for-ipns-publishers-with-http": [
-      "https://delegated-ipfs.dev/routing/v1/ipns"
-    ]
+  "Profiles": {
+    "browser": {
+      "PreferDelegated": ["AminoDHT", "IPNI", "IPNS", "DNS"]
+    },
+    "daemon": {
+      "PreferNative": ["AminoDHT", "IPNS", "DNS"],
```

Comment:
DNS seems a little awkward here in how it plays interplays with the DNS resolvers

Reply:
This way Kubo can ignore `https://delegated-ipfs.dev/dns-query` because it has own native DNS from OS, and still apply DNSResolvers for special TLDs.

We could remove it from autoconfig, but then Verified Fetch and inbrowser.link will have to hardcode `https://delegated-ipfs.dev/dns-query` or `https://1.1.1.1/dns-query` as the default resolver for ICANN TLDs.

Reply:
6f83bc3ba21c1046a2b224a5bfe87c8a6786f76d